src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs

```diff
@@ -63,7 +63,10 @@ private void HandleSyntaxTree(SyntaxTreeAnalysisContext context)
             }
 
             var expectedFileHeader = fileHeaderTemplate.Replace("{fileName}", Path.GetFileName(tree.FilePath));
-            if (!CompareCopyrightText(expectedFileHeader, fileHeader.CopyrightText))
+            if (!CompareCopyrightText(expectedFileHeader, fileHeader.CopyrightText) &&
+                // Compare the current fileHeader with the expected file header, assuming the expected file header is written as a comment
+                // e.g. file_header_template = /* Copyright */
+                !CompareCopyrightText(expectedFileHeader, root.GetText().GetSubText(fileHeader.GetHeaderLocation(tree).SourceSpan).ToString()))
```

Comment:
instead of root.GetText(), please use `tree.GetText(cancellationToken)` (and do this into a local).

Reply:
also, then just do `.GetSubText(fileHeader.Span)`

Reply:
>  tree.GetText(cancellationToken) (and do this into a local).

I didn't put it into a local, because the call is in the short circuit `&&` branch (see 099dcdd4 for details).